src/Sulu/Bundle/ContentBundle/Twig/ContentTwigExtension.php

```diff
@@ -138,7 +138,10 @@ public function needsAddButtonFunction(PropertyInterface $property)
      */
     public function isMultipleTest($property)
     {
-        return $property->getMinOccurs() > 1;
+        return
+            !is_null($property->getMaxOccurs()) ||
+            $property->getMinOccurs() > 1 ||
+            $property->getMinOccurs() < $property->getMaxOccurs();
```

Comment:
Have tested it, especially the case `minOccurs = 0 && maxOccurs = 1` makes problems, but I've structured the code a bit differently:

```php
        $minOccurs = $property->getMinOccurs();
        $maxOccurs = $property->getMaxOccurs();

        if (is_null($minOccurs) && is_null($maxOccurs)) {
            // if no occurs attributes are set it defaults to false
            return false;
        }

        if ($minOccurs < $maxOccurs) {
            // this one is especially for the case of minOccurs = 0 and maxOccurs = 1
            return true;
        }

        return $minOccurs > 1 || $maxOccurs > 1;
```

I think it catches all the examples you've listed here. The only problem is, that when you have an item with only minOccurs but no maxOccurs set you can't add any items. I'll also look into that issue now.